sys/linux/bpf.txt

```diff
@@ -96,10 +96,18 @@ bpf_map_lookup_arg {
 	flags	flags[bpf_lookup_flags, int64]
 }
 
+bpf_map_update_val [
+	buf	array[int8]
+	udp	sock_udp
+	udp6	sock_udp6
+	tcp	sock_tcp
+	tcp6	sock_tcp6
```

Comment:
If I remember correctly, this means that syzkaller will select each value uniformly at random. Is that what we want? Maybe the socket fds should be in a sub-union? On the other hand, I'm not super sure fuzzing random map values (first item) is very useful anyway.

We should probably have `fd_bpf` for prog arrays in here as well. Maybe best left to some other PR until we can check that that would increase coverage as well.

Reply:
> If I remember correctly, this means that syzkaller will select each value uniformly at random.

Correct. But...

> On the other hand, I'm not super sure fuzzing random map values (first item) is very useful anyway.

Yes. I would assume size of the map key can make difference, and using socket fd's can produce way more interesting interactions. So I would leave it as is.